core/cloudflow-operator/src/main/scala/cloudflow/operator/event/ConfigInputChangeEvent.scala

```diff
@@ -0,0 +1,259 @@
+            Action.create(outputSecret, secretEditor)
+          }
+
+        case _ ⇒ Nil // app could not be found, do nothing.
+      }
+      .mapConcat(_.toList)
+
+  def getConfigFromSecret(configInputChangeEvent: ConfigInputChangeEvent, system: ActorSystem) = {
+    val secret: Secret = configInputChangeEvent.watchEvent._object
+    secret.data
+      .get(ConfigInputChangeEvent.SecretDataKey)
+      .map { bytes =>
+        Try(ConfigFactory.parseString(new String(bytes, StandardCharsets.UTF_8))).toOption.getOrElse {
+          system.log.error(
+            s"Detected input secret '${secret.metadata.name}' contains invalid configuration data, IGNORING configuration."
+          )
+          ConfigFactory.empty()
+        }
+      }
+      .getOrElse {
+        system.log.error(
+          s"Detected input secret '${secret.metadata.name}' does not have data key '${ConfigInputChangeEvent.SecretDataKey}', IGNORING configuration."
+        )
+        ConfigFactory.empty()
+      }
+  }
+
+  def streamletConfigPath(streamletName: String)        = s"cloudflow.streamlets.$streamletName"
+  def streamletRuntimeConfigPath(streamletName: String) = s"cloudflow.streamlets.$streamletName.config"
+  def globalRuntimeConfigPath(runtime: String)          = s"cloudflow.runtimes.$runtime.config"
+
+  def getStreamletConfig(streamletName: String, appConfig: Config, runtimeConfig: Config) = {
+    val path = streamletConfigPath(streamletName)
+
+    Try(appConfig.getConfig(path)).toOption
+      .getOrElse(ConfigFactory.empty())
+      .atPath(path)
+      .withFallback(runtimeConfig)
+  }
+
+  def getStreamletRuntimeConfig(runtime: String, streamletName: String, appConfig: Config) =
+    Try(appConfig.getConfig(globalRuntimeConfigPath(runtime))).toOption
+      .getOrElse(ConfigFactory.empty())
+      .atPath(streamletRuntimeConfigPath(streamletName))
+
+  def moveConfigParameters(config: Config, streamletName: String): Config = {
+    val key                        = streamletConfigPath(streamletName)
+    val configParametersKey        = "config-parameters"
+    val absoluteConfigParameterKey = s"$key.$configParametersKey"
+    val configParametersSection = Try(Some(config.getConfig(absoluteConfigParameterKey)))
+      .getOrElse(None)
+
+    configParametersSection
+      .map { c =>
+        val adjustedConfigParametersConfig = c.atPath(key)
+        config.withoutPath(absoluteConfigParameterKey).withFallback(adjustedConfigParametersConfig)
+      }
+      .getOrElse(config)
+  }
+
+  def mergeRuntimeConfigToRoot(config: Config, streamletName: String): Config = {
+    val streamletKey = streamletConfigPath(streamletName)
+
+    val configKey                = "config"
+    val absoluteRuntimeConfigKey = s"$streamletKey.$configKey"
+    val runtimeConfigSection = Try(Some(config.getConfig(absoluteRuntimeConfigKey)))
+      .getOrElse(None)
+    // removing 'config' section and move it contents in the root of the config (akka, spark, flink, etc).
```

Comment:
```suggestion
    // removing 'config' section and move it contents in the root of the config (akka, spark, flink, etc).
```
// removing 'config' section and move its contents in the root of the config (akka, spark, flink, etc).